src/plexil-adapter/OwInterface.h

```diff
@@ -97,6 +103,44 @@ class OwInterface
                       double position, double velocity,
                       double current, double goal,
                       const ros::Time& start);
+  void systemFaultMessageCallback (const ow_faults::SystemFaults::ConstPtr&);
+  void armFaultCallback (const ow_faults::ArmFaults::ConstPtr&);
+  void powerFaultCallback (const ow_faults::PowerFaults::ConstPtr&);
+  void antennaFaultCallback (const ow_faults::PTFaults::ConstPtr&);
+  bool checkFaultMessages(std::string fault_component, 
+                                        uint32_t msg_val, 
+                                        std::string key, 
+                                        uint32_t value, 
+                                        bool b );
+
+//////////////////// FAULTS FOR SYSTEM LEVEL ////////////////////////
+
+  std::map<std::string,std::pair<uint64_t, bool>> systemErrors = 
```

Comment:
I would have made these maps static variables in the .cpp file.  Was there a certain reason to make them class members, and (@Samahu ) is initializing maps in the header file idiomatic?  I always initialize member maps and other structures in the .cpp file.

Reply:
We could make them static. I chose to make them a part of the class such that if ever we needed to recall this anywhere else in autonomy in the future.

Reply:
I placed the inifliazation in the .h folder to make the cpp cleaner since the values associated with each failure type would remain for the most part constant and the booleans are the only thing being updated if faults occur.